Makefile

```diff
@@ -0,0 +1,70 @@
+ifndef VERBOSE
+.SILENT:
+endif
+
+.PHONY: development
+development:
+	$(eval DEPLOY_ENV=development)
+
+.PHONY: staging
+staging:
+	$(eval DEPLOY_ENV=staging)
+
+.PHONY: sandbox
+sandbox:
+	$(eval DEPLOY_ENV=sandbox)
+
+.PHONY: production
+production:
+	$(eval DEPLOY_ENV=production)
+	$(if $(CONFIRM_PRODUCTION), , $(error Production can only run with CONFIRM_PRODUCTION))
+
+cpd_ecf:
+	$(eval include global_config/cpd_ecf_domain.sh)
+
+set-azure-account:
+	echo "Logging on to ${AZ_SUBSCRIPTION}"
+	az account set -s $(AZ_SUBSCRIPTION)
+
+set-azure-resource-group-tags: ##Tags that will be added to resource group on its creation in ARM template
+	$(eval RG_TAGS=$(shell echo '{"Portfolio": "Early Years and Schools Group", "Parent Business":"Teacher Training and Qualifications", "Product" : "Early Careers Framework", "Service Line": "Teaching Workforce", "Service": "Teacher services", "Service Offering": "Manage training for early career teachers", "Environment" : "$(ENV_TAG)"}' | jq . ))
+
+set-azure-template-tag:
+	$(eval ARM_TEMPLATE_TAG=1.1.0)
+
+set-production-subscription:
+	$(eval AZ_SUBSCRIPTION=s189-teacher-services-cloud-production)
+
+set-what-if:
+	$(eval WHAT_IF=--what-if)
+
+check-auto-approve:
+	$(if $(AUTO_APPROVE), , $(error can only run with AUTO_APPROVE))
+
+domain-azure-resources: set-azure-account set-azure-template-tag set-azure-resource-group-tags
+	az deployment sub create -l "UK South" --template-uri "https://raw.githubusercontent.com/DFE-Digital/tra-shared-services/${ARM_TEMPLATE_TAG}/azure/resourcedeploy.json" \
+		--name "${DNS_ZONE}domains-$(shell date +%Y%m%d%H%M%S)" --parameters "resourceGroupName=${RESOURCE_PREFIX}-${DNS_ZONE}domains-rg" 'tags=${RG_TAGS}' \
+			"tfStorageAccountName=${RESOURCE_PREFIX}${DNS_ZONE}domainstf" "tfStorageContainerName=${DNS_ZONE}domains-tf"  "keyVaultName=${RESOURCE_PREFIX}-${DNS_ZONE}domains-kv" ${WHAT_IF}
+
+validate-domain-resources: set-what-if domain-azure-resources # make cpd_ecf validate-domain-resources AUTO_APPROVE=1
+
+deploy-domain-resources: check-auto-approve domain-azure-resources # make cpd_ecf deploy-domain-resources AUTO_APPROVE=1
+
+domains-infra-init: set-production-subscription set-azure-account
```

Comment:
cpd_ecf (or "load_domain_config") can be called here so there is no need to call it on command line